File: tests/integration/features/flat-lists/flat-lists.cs
// Principle #12: parameters / arguments / generics / accessors render as
// flat siblings with field attributes. No <parameters>/<accessor_list>
// wrapper nesting.

class FlatLists
{
    public T First<T, U>(T a, U b, int c) where T : class
    {
        return a;
    }

    public int Count { get; set; }

    public void Caller()
    {
        First<string, int>("x", 1, 2);
    }
}

File: tests/integration/features/flat-lists/flat-lists.rs
// Principle #12: flat parameters / arguments / generics. No wrapping
// <parameters>/<arguments>/<type_parameters> element — children are
// direct siblings with field attributes.

fn first<T, U: Clone>(a: T, b: U, c: i32) -> T {
    a
}

fn caller() {
    first::<String, i32>(String::from("x"), 1, 2);
}
